src/GrasshopperTeklaDrawingLink/Components/Text/CreateTextComponent.cs

```diff
@@ -0,0 +1,263 @@
+
+            var insertedTexts = new TSD.Text[textNumber];
+            if(leaderLineEndPoints.Any()) {
+                CreateTextWithLeader(view,
+                    insertionPoints,
+                    leaderLineEndPoints,
+                    textElements,
+                    attributes,
+                    textNumber,
+                    insertedTexts);
+            }
+            else {
+                CreateTextWithoutLeader(view,
+                    insertionPoints,
+                    textElements,
+                    attributes,
+                    textNumber,
+                    insertedTexts);
+            }
+            DrawingInteractor.CommitChanges();
+            dA.SetDataList(ParamInfos.Text.Name, insertedTexts);
+            return insertedTexts;
+        }
+
+        private void CreateTextWithoutLeader(View view, List<Point3d> insertionPoints, List<string> textElements, List<TSD.Text.TextAttributes> attributes, int textNumber, TSD.Text[] insertedTexts) {
+            for(int i = 0; i<textNumber; i++) {
+                var text = InsertText(
+                    view,
+                    attributes.ElementAtOrLast(i),
+                    insertionPoints.ElementAtOrLast(i),
+                    textElements.ElementAtOrLast(i)
+              );
+
+                insertedTexts[i]=text;
+            }
+        }
+
+        private void CreateTextWithLeader(View view, List<Point3d> insertionPoints, List<Point3d> leaderLineEndPoints, List<string> textElements, List<TSD.Text.TextAttributes> attributes, int textNumber, TSD.Text[] insertedTexts) {
+            for(int i = 0; i<textNumber; i++) {
+                var text = InsertText(
+                    view,
+                    attributes.ElementAtOrLast(i),
+                    insertionPoints.ElementAtOrLast(i),
+                    leaderLineEndPoints.ElementAtOrLast(i),
+                    textElements.ElementAtOrLast(i)
+              );
+
+                insertedTexts[i]=text;
+            }
+        }
+
+        /// <summary>
+        /// Compares the attributes from files and components and returns the attributes to be inserted. 
+        /// As long as the component attribute is not the default value defined in <see cref="_propertiesWithDefaultValues"/>
+        /// assign its value.
+        /// </summary>
+        /// <param name="attributesFromFile"></param>
+        /// <param name="attributesFromComponent"></param>
+        /// <returns></returns>
+        private static List<TSD.Text.TextAttributes> ProcessAttributes(List<string> attributesFromFile, List<TSD.Text.TextAttributes> attributesFromComponent) {
+            var attributes = new List<TSD.Text.TextAttributes>();
+            bool attributesIsNull = attributesFromComponent is null;
+            bool attributeFilesIsEmpty = !attributesFromFile.Any();
+            byte flag = (byte)((attributesIsNull ? 1 : 0)<<1|(attributeFilesIsEmpty ? 1 : 0));
+
+
+            switch(flag) {
```

Comment:
_Here be Dragons_ :)
I appreciate that you gave the users ability to provide attribute file name in this place. But it leads to some questions and maybe hidden bugs/misunderstandings. You have created TextAttributesComponent - great builder for all of those frame types/colors/angles/etc. In TextAttributesComponent you provided the ability to specify ready-to-consume attributes file. That's great, 95% of users will consume this component by attribute file. 

So let's be strong-typed and enforce users to always use TextAttributesComponent before CreateTextComponent. It will decrease complexity of this part of the codebase and provide clear rules for consumers - always create attributes before going with real text.